DataLink.tex

```diff
@@ -981,6 +992,40 @@ \subsection{Input PARAMs}
 Services \rfcshould\ respond with an informative error message if the input is not adequate to
 perform the operations(s).
 
+\subsection{Service self-description}
+\label{sec:selfDescribingExample}
+A service may include a service descriptor that describes itself with
+its normal output.
+In that case the utype ``adhoc:this'' indicates the self-describing
+nature of the service descriptor.
+This convention makes finding the self-description unambiguous in
+cases where the output also contains other service descriptors.
+
+This usage is comparable to prototype work on S3
+(see \citet{note:s3})
+and when combined with calling a service with no input parameters
+(e.g., as allowed in \ref{sec:resourceId}),
+and/or with the DALI \texttt{MAXREC=0} convention,
+will make it easy for clients to obtain a
+description of both standard and custom features.
+
+For backward compatibility with DataLink 1.0 and SIA 2.0, client software
+conforming to the present recommendation should also treat elements of
+the form \verb|<RESOURCE type="meta" utype="adhoc:service" name="this"/>|
+as self-descriptions, equivalent to
+\verb|<RESOURCE type="meta" utype="adhoc:this" name=""/>|.
+A conforming client should treat the provision of more than one
+self-description \texttt{<RESOURCE>} element as an error, except that if a
+service provides exactly one of each of the present (DataLink 1.1 and
+beyond) and DataLink 1.0 styles, the client may silently ignore the
+DataLink 1.0 style instance.
+
+The output of a \blinks\ endpoint with no input ID would include the
+self-describing service descriptor and an empty results table.
+
+The output of a \blinks\ endpoint with a single input ID and no other parameters provided \rfcshould\ include the self-describing service descriptor with  input paramaters limits and optional values appropriate for this ID. 
+
+
```

Comment:
I'm still unconvinced this belongs into this spec; do we have an implementation? Or even more convincincly, anything that wouldn't work without it?

You see, the main complaint about datalink is that there's too many different ways to use it and it's hard to understand what to do when and what the minimal feature set is that a client needs to support.  With this section, I'm wondering myself: Do I need to do anything in pyVO?  And I frankly don't know.

Reply:
The self description as defined above line 1025 is very valuable, but I agree that there is no strong need for specifying the case where a DL service is invoked with an `obsid` but without parameters. This would requires a complex logic able to get the ranges of the parameter values for a specific dataset, thus the investment should worth it.

Reply:
I removed line 1026,  as Markus and Laurent suggest, although I think we have to provide solutions when services are accessed directly from discovery in e.g. the registry and may use IDs obtained from papers (e.g. DOIs). But I think if people would like to provide <VALUES> adapted for such IDS in their implementation this will not be considered as breaking the spec. But normative part has been cleaned up.